cluster-autoscaler/core/test/common.go

```diff
@@ -305,3 +337,67 @@ func NewBackoff() backoff.Backoff {
 	return backoff.NewIdBasedExponentialBackoff(5*time.Minute, /*InitialNodeGroupBackoffDuration*/
 		30*time.Minute /*MaxNodeGroupBackoffDuration*/, 3*time.Hour /*NodeGroupBackoffResetTimeout*/)
 }
+
+// To implement expander.Strategy, BestOption method must have a struct receiver.
+// This prevents it from modifying fields of reportingStrategy, so we need a thin
+// pointer wrapper for mutable parts.
+type expanderResults struct {
+	inputOptions []GroupSizeChange
+}
+
+// MockReportingStrategy implements expander.Strategy
+type MockReportingStrategy struct {
+	defaultStrategy expander.Strategy
+	optionToChoose  *GroupSizeChange
+	t               *testing.T
+	results         *expanderResults
+}
+
+// NewMockRepotingStrategy creates an expander strategy with reporting and mocking capabilities.
+func NewMockRepotingStrategy(t *testing.T, optionToChoose *GroupSizeChange) *MockReportingStrategy {
```

Comment:
I agree, but it's passed by reference because it can be null. If you don't care about the `optionToChoose`, one will be [picked according to the default strategy](https://github.com/pmendelski/autoscaler/blob/934296343a7f8113c49dac9472603023f303f0e1/cluster-autoscaler/core/test/common.go#L376).

Reply:
Ah, good point, let's leave as is then.